samples/matter/thermostat/src/app_task.cpp

```diff
@@ -0,0 +1,549 @@
+		event.UpdateLedStateEvent.LedWidget->UpdateState();
+	}
+}
+
+void AppTask::UpdateStatusLED()
+{
+	/* Update the status LED.
+	 *
+	 * If IPv6 networking and service provisioned, keep the LED On constantly.
+	 *
+	 * If the system has BLE connection(s) uptill the stage above, THEN blink the LED at an even
+	 * rate of 100ms.
+	 *
+	 * Otherwise, blink the LED for a very short time. */
+	if (sIsNetworkProvisioned && sIsNetworkEnabled) {
+		sStatusLED.Set(true);
+	} else if (sHaveBLEConnections) {
+		sStatusLED.Blink(LedConsts::StatusLed::Unprovisioned::kOn_ms,
+				 LedConsts::StatusLed::Unprovisioned::kOff_ms);
+	} else {
+		sStatusLED.Blink(LedConsts::StatusLed::Provisioned::kOn_ms, LedConsts::StatusLed::Provisioned::kOff_ms);
+	}
+}
+
+void AppTask::ChipEventHandler(const ChipDeviceEvent *event, intptr_t /* arg */)
+{
+	switch (event->Type) {
+	case DeviceEventType::kCHIPoBLEAdvertisingChange:
+		sHaveBLEConnections = ConnectivityMgr().NumBLEConnections() != 0;
+		UpdateStatusLED();
+		break;
+#if defined(CONFIG_NET_L2_OPENTHREAD)
+	case DeviceEventType::kDnssdInitialized:
+#if CONFIG_CHIP_OTA_REQUESTOR
+		InitBasicOTARequestor();
+#endif /* CONFIG_CHIP_OTA_REQUESTOR */
+		break;
+	case DeviceEventType::kThreadStateChange:
+		sIsNetworkProvisioned = ConnectivityMgr().IsThreadProvisioned();
+		sIsNetworkEnabled = ConnectivityMgr().IsThreadEnabled();
+#elif defined(CONFIG_CHIP_WIFI)
+	case DeviceEventType::kWiFiConnectivityChange:
+		sIsNetworkProvisioned = ConnectivityMgr().IsWiFiStationProvisioned();
+		sIsNetworkEnabled = ConnectivityMgr().IsWiFiStationEnabled();
+#if CONFIG_CHIP_OTA_REQUESTOR
+		if (event->WiFiConnectivityChange.Result == kConnectivity_Established) {
+			InitBasicOTARequestor();
+		}
+#endif /* CONFIG_CHIP_OTA_REQUESTOR */
+#endif
+		UpdateStatusLED();
+		break;
+	default:
+		break;
+	}
+}
+
+
+void AppTask::CancelTimer()
+{
+	k_timer_stop(&sFunctionTimer);
+	mFunctionTimerActive = false;
+}
+
+void AppTask::StartTimer(uint32_t timeoutInMs)
+{
+	k_timer_start(&sFunctionTimer, K_MSEC(timeoutInMs), K_NO_WAIT);
+	mFunctionTimerActive = true;
+}
+
+
+void AppTask::PostEvent(const AppEvent &event)
+{
+	if (k_msgq_put(&sAppEventQueue, &event, K_NO_WAIT) != 0) {
+		LOG_INF("Failed to post event to app task event queue");
+	}
+}
+
+void AppTask::DispatchEvent(const AppEvent &event)
+{
+	if (event.Handler) {
+		event.Handler(event);
+	} else {
+		LOG_INF("Event received with no handler. Dropping event.");
+	}
+}
+
+
+void AppTask::IdentifyStartHandler(Identify *)
+{
+	AppEvent event;
+	event.Type = AppEventType::IdentifyStart;
+	event.Handler = [](const AppEvent &) { sIdentifyLED.Blink(LedConsts::kIdentifyBlinkRate_ms);};
+	PostEvent(event);
+}
+
+void AppTask::IdentifyStopHandler(Identify *)
+{
+	AppEvent event;
+	event.Type = AppEventType::IdentifyStop;
+	event.Handler = [](const AppEvent &) { sIdentifyLED.Set(false); };
+	PostEvent(event);
+}
+
+void AppTask::UpdateThermostatStatus(void)
```

Comment:
It's not really updating anything, it just logs the thermostat status, so either change the naming or just use `TemperatureManager::Instance().LogThermostatstatus();` directly. when needed.